src/System.Management.Automation/utils/StructuredTraceSource.cs

```diff
@@ -848,20 +872,44 @@ internal void TraceLockReleased(string lockName)
         /// <param name="warningMessageFormat">
         /// The format string containing the error message
         /// </param>
-        /// <param name="args">
-        /// The arguments for the format string
-        /// </param>
+        internal void TraceWarning(string warningMessageFormat)
+        {
+            if ((_flags & PSTraceSourceOptions.Warning) != PSTraceSourceOptions.None)
+            {
+                FormatOutputLine(
+                    PSTraceSourceOptions.Warning,
+                    warningFormatter,
+                    warningMessageFormat);
+            }
+        }
+
         internal void TraceWarning(
             string warningMessageFormat,
-            params object[] args)
```

Comment:
Why do you remove the `params object[] args` overloads? These methods could be called with more arguments.

Reply:
It is internal methods and we can add new overloads in any time. 
If we leave `params object[] args` overloads we allow to use _allocating_ overloads without notification.